### PR TITLE
mavlink delete MavlinkOrbSubscription and uORB delete unused orb_stat and last update timestamp

### PR DESCRIPTION
**Pull request by dagar**
~~This isn't ready yet, but opening it now because it needs quite a lot of review.~~

This pull request replaces the mavlink module's orb subscription wrapper with direct usage of uORB::Subscriptions per mavlink stream.

On fmu-v4 (3 mavlink instances) this saves ~ 7.8 kB of ram and ~ 2.7% cpu.

``` Console
nsh> free
              total       used       free    largest
 Umem:       229440     200672      28768      22880  <-- master
 Umem:       229440     192880      36560      34144  <-- PR
```

``` Console
  PID COMMAND                   CPU(ms) CPU(%)  USED/STACK PRIO(BASE) STATE FD
    0 Idle Task                   41504 27.872   244/  512   0 (  0)  READY  3  <-- master
    0 Idle Task                   36132 30.538   244/  512   0 (  0)  READY  3  <-- PR
```

### CODE REVIEW

**Comment by dagar** [comment]
I'd like to bring this in after https://github.com/PX4/Firmware/pull/14106 settles so that we don't have to deal with the mavlink orb subscriptions that require the special "subscribe from beginning" flag.

**Comment by dagar** [comment]
https://github.com/PX4/Firmware/pull/14106 has merged and I've rebased this PR on current master.

**Comment by dagar** [comment]
I was surprised by how much memory this saved, but so far the behavior appears to be consistent with master.
###  Master `mavlink status`
Screenshot from 2020-03-12 11-03-30

###  PR `mavlink status`
Screenshot from 2020-03-12 11-04-08

**Comment by Junkim3DR** [comment]
### Tested on NXP FMUK66 v3

**Indoor test flight.**
**Mode tested**
- Stabilized

**Procedure**
- Takeoff and landing in Stabilized

**Note**
- No issue noted.

**Log**
https://review.px4.io/plot_app?log=60f88527-a901-4200-b87b-8000c67709bd


### Tested on NXP FMUK66 v3

**Indoor test flight.**
**Mode tested**
- Stabilized

**Procedure**
- Takeoff and landing in Stabilized

**Note**
- No issue noted.

**Log**
https://review.px4.io/plot_app?log=bc50f150-e44b-496c-91ce-283cccda0478

**Comment by jorge789** [comment]
### Tested on PixRacer V4

**Indoor Flight Test**
Modes Tested
Stabilized Mode: Good.

**Procedure**
Arm and Take off in stabilized mode.

**Notes**
No issues noted, good flight in general.

**Log**  https://review.px4.io/plot_app?log=d95b3692-baa7-44e3-b280-664bd4588415

### Tested on CUAV nano V5

**Indoor Flight Test**
Modes Tested
Stabilized Mode: Good.

**Procedure**
Arm and Take off in stabilized mode.

**Notes**
No issues noted, good flight in general.

**Log** https://review.px4.io/plot_app?log=97d9125c-a7ef-4691-ad0b-8101be666576

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -2623,26 +2478,24 @@ class MavlinkStreamOdometry : public MavlinkStream
 
 	unsigned get_size() override
 	{
-		return (_odom_time > 0) ? MAVLINK_MSG_ID_ODOMETRY_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+		if (_mavlink->odometry_loopback_enabled()) {
+			return _vodom_sub.advertised() ? MAVLINK_MSG_ID_ODOMETRY_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+
+		} else {
+			return _odom_sub.advertised() ? MAVLINK_MSG_ID_ODOMETRY_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+		}
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_messages.cpp]: Nice catch!

File: src/modules/mavlink/mavlink_messages.cpp [hunk 2]
```diff
@@ -4339,38 +4137,33 @@ class MavlinkStreamNavControllerOutput : public MavlinkStream
...
-		_pos_ctrl_status_sub(_mavlink->add_orb_subscription(ORB_ID(position_controller_status))),
-		_tecs_status_sub(_mavlink->add_orb_subscription(ORB_ID(tecs_status)))
+	explicit MavlinkStreamNavControllerOutput(Mavlink *mavlink) : MavlinkStream(mavlink)
 	{}
 
 	bool send(const hrt_abstime t) override
 	{
-		position_controller_status_s pos_ctrl_status = {};
-		tecs_status_s tecs_status = {};
+		if (_pos_ctrl_status_sub.updated()) {
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_messages.cpp]: I suppose this change of functionality is intended? It is now only triggered on `pos_ctrl_status` updated and not on `tecs_status` anymore.

Comment (dagar) [reply src/modules/mavlink/mavlink_messages.cpp]: Yes, the MC position controller publishes a (partially populated) `position_controller_status` now.

**Comment by dagar** [comment]
Comparison with master.

### Master
Screenshot from 2020-03-14 11-36-11

### PR
Screenshot from 2020-03-14 12-39-04

**Comment by dagar** [comment]
Comparison with master (gazebo iris after takeoff).

### Master
Screenshot from 2020-03-14 12-46-04

### PR
Screenshot from 2020-03-14 12-42-46

**Comment by dagar** [comment]
Note for later, streams set to 0.5 Hz seem to jump between 0.2 Hz and 0.8 Hz (both master and the PR).

**Pull request merged**